### PR TITLE
Merge `Drawable` Class dev-branch to main repo

### PR DESCRIPTION
**Pull request by ntut-rick**
The development branch is with @NOOBDY for #39.
However, integration needs collaboration, move it to the main repo for better work.

### CODE REVIEW

**Comment by NOOBDY** [comment]
I don't think I should review my code

**Comment by ntut-rick** [comment]
I can't review my PR, btw.

File: include/Core/Drawable.hpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+namespace Core {
+struct Matrices {
+    glm::mat2 m_Model;
+    glm::mat2 m_Projection;
+};
+
+class Drawable {
+public:
+    virtual void Draw() = 0;
+
```
Comment (Dragonfly911117) [inline include/Core/Drawable.hpp:22]: Missed virtual destructor causes warning in the future

File: include/Core/Texture.hpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    Texture &operator=(const Texture &) = delete;
+    Texture &operator=(Texture &&other);
+
+    void Bind(int slot) const;
+    void Unbind() const;
+
+    void UpdateData(unsigned int channels, int width, int height,
+                    const void *data);
+
+    GLuint GetTextureId() const { return m_TextureId; }
```
Comment (Dragonfly911117) [inline include/Core/Texture.hpp:24]: The getter should be above `Bind()`

File: include/Util/Image.hpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    glm::mat3 m_Transform;
+    std::unique_ptr<Core::Texture> m_Texture = nullptr;
+    std::unique_ptr<SDL_Surface, std::function<void(SDL_Surface *)>> m_Surface =
+        nullptr;
+
+    static constexpr int UNIFORM_SURFACE_LOCATION = 0;
+
+    static std::unique_ptr<Core::Program> s_Program;
+    static std::unique_ptr<Core::VertexArray> s_VertexArray;
+    static std::unique_ptr<Core::UniformBuffer<Core::Matrices>> s_UniformBuffer;
```
Comment (Dragonfly911117) [inline include/Util/Image.hpp:34]: > Static members should be declared above normal members.

File: include/Util/Text.hpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    std::unique_ptr<SDL_Surface, std::function<void(SDL_Surface *)>> m_Surface =
+        nullptr;
+
+    std::unique_ptr<TTF_Font, std::function<void(TTF_Font *)>> m_Font;
+
+    static constexpr int UNIFORM_SURFACE_LOCATION = 0;
+
+    static std::unique_ptr<Core::Program> s_Program;
+    static std::unique_ptr<Core::VertexArray> s_VertexArray;
+    static std::unique_ptr<Core::UniformBuffer<Core::Matrices>> s_UniformBuffer;
```
Comment (Dragonfly911117) [inline include/Util/Text.hpp:37]: > Static members should be declared above normal members.

**Review by Dragonfly911117 — CHANGES_REQUESTED** [review]

File: include/Core/Drawable.hpp (hunk 1 above)
Comment (Dragonfly911117) [reply include/Core/Drawable.hpp:22]:
```patch
Subject: [PATCH] Add a virtual destructor to Core::Drawable
---
Index: include/Core/Drawable.hpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/include/Core/Drawable.hpp b/include/Core/Drawable.hpp
--- a/include/Core/Drawable.hpp	(revision 15eb0de344348cd86453612a7b7d5d69f6267882)
+++ b/include/Core/Drawable.hpp	(date 1696221657870)
@@ -18,6 +18,7 @@
 ``` should work

File: include/Core/Drawable.hpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+namespace Core {
+struct Matrices {
+    glm::mat2 m_Model;
+    glm::mat2 m_Projection;
+};
+
+class Drawable {
+public:
+    virtual void Draw() = 0;
```
Comment (Dragonfly911117) [inline include/Core/Drawable.hpp:21]:
```suggestion
    virtual ~Drawable() = default;
    virtual void Draw() = 0;
```

**Review by Dragonfly911117 — APPROVED** [review]
LGTM

**Pull request merged**